rfcs/20230117-support-boolean-data-type/README.md

```diff
@@ -0,0 +1,305 @@
+       /// 32-bit/single-precision floating point.
+       dnnl_f32 = 3,
+       /// 32-bit signed integer.
+       dnnl_s32 = 4,
+       /// 8-bit signed integer.
+       dnnl_s8 = 5,
+       /// 8-bit unsigned integer.
+       dnnl_u8 = 6,
+       /// 64-bit/double-precision floating point.
+       dnnl_f64 = 7,
+       /// boolean data type. C++ implementation defined.
+       dnnl_boolean = 8,
+        
+       /// Parameter to allow internal only data_types without undefined
+       /// behavior. This parameter is chosen to be valid for so long as
+       /// sizeof(int) >= 2.
+       dnnl_data_type_max = 0x7fff,
+   } dnnl_data_type_t;
+   ```
+
+3. oneDNN Graph
+   oneDNN Graph needs to add `boolean` in supported data types.
+
+    ``` cpp
+    /// Data Type
+    enum class data_type {
+        undef = dnnl_data_type_undef,
+        /// 16-bit/half-precision floating point.
+        f16 = dnnl_f16,
+        /// non-standard 16-bit (bfloat16 w/ 7 bit mantissa) floating point.
+        bf16 = dnnl_bf16,
+        /// 32-bit/single-precision floating point.
+        f32 = dnnl_f32,
+        /// 32-bit signed integer.
+        s32 = dnnl_s32,
+        /// 8-bit signed integer.
+        s8 = dnnl_s8,
+        /// 8-bit unsigned integer.
+        u8 = dnnl_u8,
+        /// boolean data type. C++ implementation defined.
+        boolean = dnnl_boolean,
+    };
+    ```
+
+4. Frameworks
+   Frameworks map `bool` to oneDNN Graph `data_type::boolean` for C++ and
+   `dnnl_boolean` for C.
+
+## Add `Select` operation in oneDNN Graph operation set
+
+### Introduction
+
+Based on framework developers' requirements for landing zoon model DistillBert
+optimization, Graph compiler defines a new pattern called `MHA` which include
+`Select` operation. Before fusing `Select` in its partition, we need to add
+`Select` operation into oneDNN Graph operation set first.
+
+### Formulas
+
+The `Select` operation is a conditional operation, which returns a tensor filled
+with the elements from the second or the third inputs, depending on the
+condition (the first input) value.
+> dst[i] = cond[i] ? then[i] : else[i]
+
+### Operations
+
+#### Proposed Attributes
+
+Only one attribute for `select`operation: auto_broadcast.
+Attribute Name | Description | Value Type |Supported Values | Required or Optional
+-- | -- | --| --|--
+[auto_broadcast](@ref dnnl::graph::op::attr::auto_broadcast) | Specifies rules used for auto-broadcasting of src tensors. |string |`none`, `numpy` (default)  | Optional
+
+#### Proposed Inputs and Outputs
+
+There will be 3 input tensors and 1 output tensor for the proposed Select
+operation.
+
+##### Inputs
+
+| Index | Argument Name | Required or Optional |
+| ----- | ------------- | -------------------- |
+| 0     | `cond`        | Required             |
+| 1     | `then`        | Required             |
+| 2     | `else`        | Required             |
+@note Shapes of `cond`, `then` and `else` should match and no auto-broadcasting
```

Comment:
add a blank line above since rendered table is not correctly formated.

https://github.com/rongzha1/oneDNN/tree/rzhang/rfcs_boolean_dt_and_select_op/rfcs/20230117-support-boolean-data-type#inputs

Reply:
got it. thanks